homeassistant/components/google/calendar.py

```diff
@@ -147,77 +148,66 @@ def name(self) -> str:
         """Return the name of the entity."""
         return self._name
 
-    def _event_filter(self, event: dict[str, Any]) -> bool:
+    def _event_filter(self, event: Event) -> bool:
         """Return True if the event is visible."""
         if self._ignore_availability:
             return True
-        return event.get(TRANSPARENCY, OPAQUE) == OPAQUE
+        return event.transparency == OPAQUE
 
     async def async_get_events(
         self, hass: HomeAssistant, start_date: datetime, end_date: datetime
     ) -> list[CalendarEvent]:
         """Get all events in a specific time frame."""
         event_list: list[dict[str, Any]] = []
-        page_token: str | None = None
+
+        request = ListEventsRequest(
+            calendar_id=self._calendar_id,
+            start_time=start_date,
+            end_time=end_date,
+            search=self._search,
+        )
         while True:
             try:
-                items, page_token = await self._calendar_service.async_list_events(
-                    self._calendar_id,
-                    start_time=start_date,
-                    end_time=end_date,
-                    search=self._search,
-                    page_token=page_token,
-                )
-            except ServerNotFoundError as err:
+                result = await self._calendar_service.async_list_events(request)
+            except ApiException as err:
                 _LOGGER.error("Unable to connect to Google: %s", err)
                 return []
 
-            event_list.extend(filter(self._event_filter, items))
-            if not page_token:
+            event_list.extend(filter(self._event_filter, result.items))
+            if not result.page_token:
                 break
 
+            request.page_token = result.page_token
```

Comment:
I would recommend making an async iterable instead so the library user can simply iterate over and fetch existing pages in the result. This is what aiogoogle does.

Needing to handle page tokens is normally not interesting for the library user.